ecs.tf

```diff
@@ -59,43 +51,70 @@ resource "aws_appautoscaling_target" "app_scale_target" {
   min_capacity       = var.ecs_autoscale_min_instances
 }
 
+module "app_container_definition" {
+  source                       = "git::https://github.com/cloudposse/terraform-aws-ecs-container-definition.git?ref=tags/0.21.0"
+  container_name               = var.container_name
+  container_image              = var.docker_image
+  container_cpu                = var.container_cpu
+  container_memory             = var.container_memory
+  essential                    = true
+  readonly_root_filesystem     = false
+  environment                  = var.environment_vars
+  port_mappings                = [
+    {
+      containerPort            = var.container_port
+      hostPort                 = var.container_port
+      protocol                 = "tcp"
+    },
+  ]
+  log_configuration            = {
+    logDriver = "awslogs"
+    options   = {
+      "awslogs-group" = "/fargate/service/${var.app}-${var.environment}"
+      "awslogs-region" = "eu-west-1"
+      "awslogs-stream-prefix" = "ecs"
+    }
+    secretOptions = null
+  }
+}
+
+module "datadog_container_definition" {
+  source                       = "git::https://github.com/cloudposse/terraform-aws-ecs-container-definition.git?ref=tags/0.21.0"
+  container_name               = "datadog-agent"
+  container_image              = "datadog/agent:latest"
+  container_cpu                = "10"
+  container_memory             = "128"
+  essential                    = true
```

Comment:
I'm not sure if this should be essential as then it dying will take down the rest of the application. I took the view when setting this up that the impact of losing metrics was less than the impact of potential downtime if all the running container instances die at the same time, particularly as you can make Datadog trigger alerts on no data.

Reply:
In all the examples that Datadog gives, it is set as essential (e.g.  https://docs.datadoghq.com/resources/json/datadog-agent-ecs-fargate.json). I'm happy to remove that in this case though and see how it performs.